### PR TITLE
feat: handshake

### PR DESCRIPTION
**Pull request by pchalupa**
GH card ticket "link": #ADD LINK TO GITHUB TICKET OR ISSUE

### Description

Implemented "handshake" to example app.

### CODE REVIEW

File: apps/example/src/components/Chat.tsx [hunk 1]
```diff
@@ -4,93 +4,126 @@ import './styles.css';
...
+  typeof object === 'object';
+
+const isMessage = (message: unknown): message is Message =>
+  isObject(message) &&
+  !!message &&
+  'id' in message &&
+  typeof (message as Message).id === 'string' &&
+  'message' in message &&
+  typeof (message as Message).id === 'string';
+
```
Comment (BrantalikP) [inline apps/example/src/components/Chat.tsx:26]: This seems to be a bit overkill do not you think? I think object  === 'object' does not do much anyway. 
cannot we use something like 
```
const {message, id } = messageData
if(message && id) return true
return false
```

File: apps/example/src/components/Chat.tsx [hunk 2]
```diff
@@ -4,93 +4,126 @@ import './styles.css';
...
+const isPeer = (peer: unknown): peer is Peer =>
+  isObject(peer) &&
+  !!peer &&
+  'id' in peer &&
+  'username' in peer &&
+  typeof (peer as Peer).id === 'string';
+
```
Comment (BrantalikP) [inline apps/example/src/components/Chat.tsx:33]: same

**Review by BrantalikP — DISMISSED** [review]

File: apps/example/src/components/Chat.tsx (hunk 2 above)
Comment (pchalupa) [reply apps/example/src/components/Chat.tsx:31]: Added [zod](https://zod.dev/?id=objects) for object parsing.

File: apps/example/src/components/Chat.tsx (hunk 1 above)
Comment (pchalupa) [reply apps/example/src/components/Chat.tsx]: Added [zod](https://zod.dev/?id=objects) for object parsing.

**Review by BrantalikP — APPROVED** [review]

**Pull request merged**